### PR TITLE
Feature/users

### PR DESCRIPTION
**Pull request by MeghanaBhatManikal**
I have added User endpoints. Please review and merge.

### CODE REVIEW

File: data/users.js [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+        return await this.remove_song(objId, updateSongs);   
+    },
+
+    async check_valid(id, songId) {
+        if (!id)
+            throw `You must provide an id to search for`;
+        if (!songId)
+            throw `You must provide a songId for the song to be added`;
+        //TO-DO
+        // Verify if the song is present in teh songs collection
```
[the remark this answers is not in the document]
Comment (MeghanaBhatManikal) [reply data/users.js:273]: It is to be done once songs collection is ready. I just added it as a reminder.

[the remark this answers is not in the document]
Comment (pszenher) [reply data/users.js:273]: Perfect, we'll leave the comment in for now then

**Comment by MeghanaBhatManikal** [comment]
> The Apple ".DS_Store" file was added to the repo and should be removed before pulling in

Will remove it in my next commit, or do you want me to do it now.?

**Comment by MeghanaBhatManikal** [comment]
> The `database_proposal.pdf` and `proposal.pdf` files were removed, which is fine since we have them on Google Drive. However, if we do remove them we should probably remove the `proposal.org` file too since it doesn't really need to be stored in git.

Those files were part of .gitignore so removed it, we can check in those file and remove from the .gitignore, I would like to do it as part of next commit, if its fine. What do u think ?

**Comment by pszenher** [comment]
> > The Apple ".DS_Store" file was added to the repo and should be removed before pulling in
> 
> Will remove it in my next commit, or do you want me to do it now.?

I just committed a fix removing `.DS_Store` and adding it to the gitignore.  I've approved the PR for merge into master.

**Comment by pszenher** [comment]
> > The `database_proposal.pdf` and `proposal.pdf` files were removed, which is fine since we have them on Google Drive. However, if we do remove them we should probably remove the `proposal.org` file too since it doesn't really need to be stored in git.
> 
> Those files were part of .gitignore so removed it, we can check in those file and remove from the .gitignore, I would like to do it as part of next commit, if its fine. What do u think ?

It would be best to simply remove the proposal directory from git entirely, as it adds noise to the directory layout of the project code.  Furthermore, the database proposal is already out of date due to the changes we made a few weeks ago.  Google Drive is definitely the better medium to manage project documents.

I'll create a new branch and PR specifically for the deletion of the proposal directory and its remaining contents.  I'll merge your PR in tonight to make sure master stays as up to date as possible.

**Pull request merged**